diff --git a/package.json b/package.json
index ef8c6cbdfed28efa46496c4703d4f8eaa8feb7e2..e2666894aede577957f2e9619e5abb479ed671e8 100644
--- a/package.json
+++ b/package.json
@@ -30,13 +30,13 @@
     "css-loader": "^5.2.6",
     "csso": "^4.2.0",
     "debug": "^4.3.2",
+    "ember-style-loader": "github:mainmatter/ember-style-loader#main",
     "fs-extra": "^9.1.0",
     "jsdom": "^16.6.0",
     "lodash": "^4.17.21",
     "mini-css-extract-plugin": "^2.5.3",
     "semver": "^7.3.5",
     "source-map-url": "^0.4.1",
-    "style-loader": "^2.0.0",
     "terser": "^5.7.0",
     "thread-loader": "^3.0.4"
   },
diff --git a/src/ember-webpack.js b/src/ember-webpack.js
index d1368ccfd60f39a19b3db01991cb6af0d85d94b4..15d30ac8b141c8a6ff781360104eb0a15c8b2fbe 100644
--- a/src/ember-webpack.js
+++ b/src/ember-webpack.js
@@ -233,7 +233,7 @@ const Webpack = (_a = class Webpack {
                         'thread-loader': require.resolve('thread-loader'),
                         'babel-loader-8': require.resolve('@embroider/babel-loader-8'),
                         'css-loader': require.resolve('css-loader'),
-                        'style-loader': require.resolve('style-loader'),
+                        'ember-style-loader': require.resolve('ember-style-loader'),
                     },
                 },
             };
@@ -489,8 +489,7 @@ const Webpack = (_a = class Webpack {
                 // fastboot), we can use style-loader because it's fast
                 return {
                     loaders: [
-                        { loader: 'style-loader', options: { injectType: 'styleTag', ...this.extraStyleLoaderOptions } },
-                        cssLoader,
+                        { loader: 'ember-style-loader', options: { assetPath: '/assets'}},
                     ],
                     plugins: [],
                 };